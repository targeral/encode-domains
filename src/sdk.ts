import path from 'path';
import { transformAsync } from '@babel/core';
import { fs } from '@modern-js/utils';

export const transformCodeAsync = async (code: string) => {
  const result = await transformAsync(code, {
    // eslint-disable-next-line @typescript-eslint/no-require-imports
    plugins: [require('./babel-plugin')],
  });

  if (result === null || !result.code) {
    throw new Error('Could not transform file');
  }

  return result.code;
};

export const transformFileAsync = async (
  targetFilePath: string,
  outputFilePath?: string,
) => {
  const content = await fs.readFile(targetFilePath, 'utf-8');
  const outputCode = await transformCodeAsync(content);

  const filePath =
    outputFilePath ?? path.join(process.cwd(), './encode-domain');

  fs.ensureFileSync(filePath);
  fs.writeFileSync(filePath, outputCode);
};

export const transformUrl = (str: string) => {
  return `atob(${btoa(str)})`;
};
