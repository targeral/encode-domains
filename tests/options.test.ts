import { transformCodeAsync } from '@/sdk';

describe('Options cases', () => {
  test(`const urlObj = {a: '//example.com'} with urls option`, async () => {
    const code = `const urlObj = {a: '//example.com'}`;
    const s = await transformCodeAsync(code, { urls: ['//example.com'] });
    expect(s).toMatchSnapshot();
  });

  test(`const urlObj = {a: '//example.com'} with ignore option`, async () => {
    const code = `const urlObj = {a: 'http://example.com', b: 'http://demo.com'}`;
    const s = await transformCodeAsync(code, {
      ignore: value => /demo.com$/.test(value),
    });
    expect(s).toMatchSnapshot();
  });

  test(`const urlObj = {a: '//example.com'} with match option`, async () => {
    const code = `const urlObj = {a: '//example.com'}`;
    const s = await transformCodeAsync(code, {
      match: value => /example.com$/.test(value),
    });
    expect(s).toMatchSnapshot();
  });
});
