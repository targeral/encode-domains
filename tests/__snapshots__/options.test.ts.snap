// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options cases const urlObj = {a: '//example.com'} with ignore option 1`] = `
"const urlObj = {
  a: atob("aHR0cDovL2V4YW1wbGUuY29t"),
  b: 'http://demo.com'
};"
`;

exports[`Options cases const urlObj = {a: '//example.com'} with match option 1`] = `
"const urlObj = {
  a: atob("Ly9leGFtcGxlLmNvbQ==")
};"
`;

exports[`Options cases const urlObj = {a: '//example.com'} with urls option 1`] = `
"const urlObj = {
  a: atob("Ly9leGFtcGxlLmNvbQ==")
};"
`;

exports[`Options cases match option 1`] = `
"const lr = {
  localhost: "local",
  [atob("aGctYm9lLmJ5dGVkYW5jZS5uZXQ=")]: "hg-boe"
};"
`;
